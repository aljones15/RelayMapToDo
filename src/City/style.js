const K_WIDTH = 40;
const K_HEIGHT = 40;
import { StyleSheet } from 'aphrodite';

// initially any map object has left top corner at lat lng coordinates
// it's on you to set object origin to 0,0 coordinates

export const style = StyleSheet.create({
  Icon: {
    position: 'absolute',
    width: K_WIDTH,
    height: K_HEIGHT,
    left: -K_WIDTH / 2,
    top: -K_HEIGHT / 2,
    border: '5px solid rgba(0,0,0,0)',
    borderRadius: K_HEIGHT,
    backgroundColor: 'rgba(0,0,0,0.5)',
    textAlign: 'center',
    color: '#3f51b5',
    fontSize: 16,
    fontWeight: 'bold',
    padding: 4,
    cursor: 'context-menu',
  },
  List: {
    position: 'absolute',
    width: K_WIDTH * 10,
    height: K_HEIGHT * 10,
    left: (-K_WIDTH * 10) / 2,
    top: (-K_HEIGHT * 10) / 2,
    backgroundColor: 'rgba(0,0,0,0.2)',
    overflow: 'auto'
  },
  Container: {
    display: 'flex',
    flexGrow: 1,
    flexDirection: 'column',
    position: 'relative',
    height: '100%',
    width: '100%',
    backgroundColor: 'rgba(0,0,0,0.5)',
    overflow: 'auto'
  },
  CloseBar: {
    display: 'flex',
    flexGrow: 1,
    flexShrink: 1,
    height: '2rem',
    maxHeight: '2rem',
    width: K_WIDTH * 10,
    maxWidth: '100%',
    backgroundColor: 'rgba(10,10,10,0.4)',
    cursor: 'context-menu',
    fontSize: '1rem',
    color: '#FFFFFF',
    alignItems: 'center',
    justifyContent: 'center',
  }
});
