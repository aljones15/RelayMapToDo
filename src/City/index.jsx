import React from 'react';
import { style } from './style';
import ToDoList from '../ToDoList/index.jsx';
import {graphql, createFragmentContainer } from 'react-relay';
import { css } from 'aphrodite';

class CityView extends React.Component{
  constructor(props){
    super(props);
    this.state = {open: false};
  }
  toggleToDoList(_){
    this.setState((pstate, props) => { return {open: !pstate.open} });
  }
  toggleStyle(){
    if(this.state.open){
      return css(style.List) + " City";
    } else {
      return css(style.Icon) + " City";
    }
  }
  toggle(){
    if(this.state.open){ 
      return(
	<div className={css(style.container) + ' ToDoList'}>
	  <div className={css(style.CloseBar) + ' ToDoListClose'}
            onClick={this.toggleToDoList.bind(this)}>Close</div>
          <ToDoList city_id={this.props.item._id} relay={this.props.relay} />
	</div>
      );
    } else {
      return(
        <img src='img/list/list-flat.svg' 
          onError={() => this.src='img/list/32x32.png'} 
          onClick={this.toggleToDoList.bind(this)}/>
      );
    }
  }
  render(){
    return(
      <div 
        className={this.toggleStyle()} 
      >
      {
        this.toggle()
      }  
      </div>   
    ); 
   }
}

const City = createFragmentContainer(
  CityView,
  graphql`
    fragment City_item on City {
      _id 
    }
  `,
);

export default City;
