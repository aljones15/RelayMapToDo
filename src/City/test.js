import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import City from './index';

const City_item = {
  _id: 1
}

describe('map should', () => {
  const elJson = ReactTestRenderer.create(
    <City item={City_item} />
   ).toJSON();
  it('match a snapshot', () => {
    expect(elJson).toMatchSnapshot();
  });
});
