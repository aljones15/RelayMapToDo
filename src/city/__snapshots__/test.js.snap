// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`map should match a snapshot 1`] = `
<div
  className="City"
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,0.5)",
      "border": "5px solid rgba(0,0,0,0)",
      "borderRadius": 40,
      "color": "#3f51b5",
      "cursor": "context-menu",
      "fontSize": 16,
      "fontWeight": "bold",
      "height": 40,
      "left": -20,
      "padding": 4,
      "position": "absolute",
      "textAlign": "center",
      "top": -20,
      "width": 40,
    }
  }
>
  <img
    onClick={[Function]}
    onError={[Function]}
    src="img/list/list-flat.svg"
  />
</div>
`;
